### PR TITLE
5% fee not always sent

### PR DESCRIPTION
**Pull request by masparrow**
Resolves https://github.com/brave/brave-browser/issues/7425

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:

As per https://github.com/brave/brave-core/tree/issues/7425

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: components/brave_rewards/browser/rewards_service_impl.cc [hunk 1]
```diff
@@ -3912,7 +3912,7 @@ void RewardsServiceImpl::SetTransferFee(
     new_wallet = wallet->Clone();
   }
 
-  auto* fees = wallet->FindListKey("transfer_fees");
+  auto* fees = wallet->FindDictKey("transfer_fees");
```
Comment (tmancey) [inline components/brave_rewards/browser/rewards_service_impl.cc]: fees will be a `nullptr` if `transfer_fees` does not exist in the dictionary. Not an issue for this PR but we should pick these up in future PRs.

Comment (NejcZdovc) [reply components/brave_rewards/browser/rewards_service_impl.cc]: we check fees object on line 3917

**Comment by masparrow** [comment]
Adding browser test by request @NejcZdovc

File: components/brave_rewards/browser/rewards_service_impl.cc (hunk 1 above)
Comment (tmancey) [reply components/brave_rewards/browser/rewards_service_impl.cc]: Damn I must get some more sleep...

**Comment by tmancey** [comment]
@NejcZdovc Failed due to test install, so up to you if you want to run CI again

**Pull request merged**